### ISSUE TITLE
Document Row syntax better

### ISSUE THREAD

**Issue by JordanMartinez** [JordanMartinez/purescript-jordans-reference#91]
```purescript
-- closed rows
type Example_of_an_Empty_Row = ()
type Example_of_a_Single_Row = (fieldName :: ValueType)
type Example_of_a_Multiple_Rows = (first :: ValueType, second :: ValueType)

-- open rows
type Example_of_Open_Rows additionalRows =
  (first :: ValueType, second :: ValueType | additionalRows)
type Example_of_Anything_and_Everything_Rows r = ( | r)
```

**Comment by JordanMartinez** [comment]
I've documented this in #92, but I'd like to introduce it better than how I've done it now. However, this will require a breaking change.

**Comment by JordanMartinez** [comment]
Closed by #95
